Update localization docs (#866)
Automattic/studio: docs/localization.md

#### Step 2: Import to GlotPress:

   1. Open [our project in GlotPress](https://translate.wordpress.com/projects/studio/).

>    2. Click the **Project actions** menu. (You should be proxied through AutoProxxy or similar way)

Reviewer: Aren't these documents public? I am not completely convinced that we should mention AutoProxxy there.

Reviewer 2: What about mentioning the requirement and not the tools?

```suggestion
   2. Click the **Project actions** menu. You must have admin permissions to view the menu.
```

Author: @sejas yeah, makes sense 👍

Reviewer 3: I think this is OK, but people could still get confused about what "admin permissions" means. We could tweak it to something like "You must be a proxied Automattician to view the menu" to make it more instantly understandable. However, I also assume a12s are typically proxied, making this documentation somewhat redundant…

Author: > However, I also assume a12s are typically proxied, making this documentation somewhat redundant…

Sometimes I forget to run AutoProxxy, since it's not critical for Studio development. And that was the case, so I could not understand where this menu is, since I had access but the menu is hidden.

Author: @sejas WDYT about `You must be a proxied Automattician to view the menu`?
I think it's indeed more direct, clear and we don't mention AutoProxxy

Reviewer 2: Sounds good to me. It's simple and direct 👍.
